src/server_manager/ui_components/outline-manual-server-entry.html

```diff
@@ -329,7 +383,17 @@
         this.$.serverConfig.value = '';
         this.showConnection = false;
         this.$.awsDropDown.hide();
-        this.$.gcpDropDown.hide();
+        // TODOBEFOREPUSH use $$ to select all drop-downs
+        this.$.gcpFirewallDropDown.hide();
+        this.$.gcpCreateServerDropDown.hide();
+      },
+      _cloudProviderChanged: function(newProvider) {
```

Comment:
This is getting a bit messy. The isCloudProvider logic is being duplicated. Do we really need a field and a getter?

Also, the overall logic is getting more complicated because of a change specific to GCP. Can we improve encapsulation so we can understand the screen without having to dig into cloud-specific details? I think having a component for each cloud option would help a lot.

Reply:
I agree.  This code will simplify a bit once I also do the AWS options, so could I punt until then so I can do it in one fell swoop?

Reply:
ok, please add a TODO

Reply:
Done in 25ee445